ArduPlane/GCS_Mavlink.cpp

```diff
@@ -791,13 +792,197 @@ MAV_RESULT GCS_MAVLINK_Plane::handle_command_int_do_reposition(const mavlink_com
+MAV_RESULT GCS_MAVLINK_Plane::handle_command_int_guided_slew_commands(const mavlink_command_int_t &packet)
+{
+
+
+  ::printf("GCS_MAVLINK_Plane::handle_command_int_guided_slew_commands .. slew packet recieved:%d\n",packet.command);
+
+  switch(packet.command) {
+    
+
+#if OFFBOARD_GUIDED == ENABLED
+    case MAV_CMD_GUIDED_CHANGE_SPEED: {
+        // command is only valid in guided
+        if (plane.control_mode != &plane.mode_guided) {
+            gcs().send_text(MAV_SEVERITY_WARNING,"TEMPORARILY_REJECTED - not in guided mode.");
+            return MAV_RESULT_TEMPORARILY_REJECTED;
+        }
+
+         // only airspeed commands are supported right now...
+        if (int(packet.param1) != SPEED_TYPE_AIRSPEED) {  // since SPEED_TYPE is int in range 0-1 and packet.param1 is a *float* this works.
+            ::printf("unsupported packet.param1 in cmd-guided-change-speed:%f\n",packet.param1);
+            return MAV_RESULT_UNSUPPORTED;
+        }
+
+         // reject airspeeds that are outside of the tuning envelope
+        if (packet.param2 > plane.aparm.airspeed_max || packet.param2 < plane.aparm.airspeed_min) {
+            gcs().send_text(MAV_SEVERITY_WARNING,"FAILED - not inside tuning envelope.");
+            return MAV_RESULT_FAILED;
+        }
+
+         // reject duplicate airspeeds
+        float new_target_airspeed_cm = packet.param2 * 100;
+//        if (new_target_airspeed_cm == plane.guided_state.target_airspeed_cm) {
+        if ( fabs(new_target_airspeed_cm - plane.guided_state.target_airspeed_cm)<0.001) { // compare if two floats are 'the same'
+            gcs().send_text(MAV_SEVERITY_WARNING,"dup arspd request");
+            return MAV_RESULT_TEMPORARILY_REJECTED;
+        }
+        plane.guided_state.target_airspeed_cm = new_target_airspeed_cm;
+        plane.guided_state.target_airspeed_time_ms = AP_HAL::millis();
+
+         if (is_zero(packet.param3)) {
+            // the user wanted /maximum acceleration, pick a large value as close enough
+            plane.guided_state.target_airspeed_accel = 1000.0f;
+        } else {
+            plane.guided_state.target_airspeed_accel = fabsf(packet.param3);
+        }
+
+         // assign an acceleration direction
+        if (plane.guided_state.target_airspeed_cm < plane.target_airspeed_cm) {
+            plane.guided_state.target_airspeed_accel *= -1.0f;
+        }
+        ::printf("ACCEPTED spd:%f -> %f\n",packet.param2,packet.param3);
+        gcs().send_text(MAV_SEVERITY_WARNING,"ACCEPTED spd");
+        return MAV_RESULT_ACCEPTED;
+    }
+
+     case MAV_CMD_GUIDED_CHANGE_ALTITUDE: {
+        // command is only valid in guided
+        if (plane.control_mode != &plane.mode_guided) {
+            gcs().send_text(MAV_SEVERITY_WARNING,"REJECTED - not in guided mode");
+            return MAV_RESULT_TEMPORARILY_REJECTED;
+        }
+
+        // command is only valid if alt is non-zero
+        if (  packet.z < 0.0001 ) {
```

Comment:
Whitespace

Reply:
Also don't reject altitudes below 0, there are valid reasons to fly below where you take off from, which I've used in the past.

Reply:
will fix on next attempt.